chain/beacon/drand/drand.go

```diff
@@ -119,16 +119,26 @@ func NewDrandBeacon(genesisTs, interval uint64) (*DrandBeacon, error) {
 }
 
 func (db *DrandBeacon) rotatePeersIndex() {
+	db.peersIndexMtx.Lock()
+	nval := rand.Intn(len(db.peers))
```

Comment:
:+1: on rand here, otherwise under some conditions we could have rolling failure where all nodes sync up on one beacon and it fails due to load and on for the next one.

Reply:
now I noticed that it is what you called "thundering herd"